### PR TITLE
Analyzer/population analysis

### PR DESCRIPTION
**Pull request by NoamShapira**

### CODE REVIEW

File: src/corona_hakab_model/analyzers/config.py [hunk 1]
```diff
@@ -0,0 +1,5 @@
+from os import path
+from project_structure import OUTPUT_FOLDER
+
+POPULATION_OUTPUT_PATH = path.join(path.dirname(path.abspath(__file__)), "../../../output")
```
Comment (bgmoshe) [inline src/corona_hakab_model/analyzers/config.py]: Why not use OUTPUT_FOLDER? I dont think we want multiple output folders, it is confusing

File: src/corona_hakab_model/analyzers/example_script.py [hunk 2]
```diff
@@ -31,3 +32,7 @@
 # you can also plot all the parameters time propagation
 plot_parameter_propagation_aggregated(mean_df=results_df_dict["mean"], std_df=results_df_dict["std"])
 
+
+# plot social circles sickness analisys
+initial_sick_data_df, final_sick_data_df, all_circles_df = load_population_data_to_dfs()
```
Comment (bgmoshe) [inline src/corona_hakab_model/analyzers/example_script.py:37]: If you do not use the initial_sick_data_df why not use python temporary variable _ instead?

File: src/corona_hakab_model/analyzers/example_script.py [hunk 3]
```diff
@@ -31,3 +32,7 @@
 # you can also plot all the parameters time propagation
 plot_parameter_propagation_aggregated(mean_df=results_df_dict["mean"], std_df=results_df_dict["std"])
 
+
+# plot social circles sickness analisys
```
Comment (bgmoshe) [inline src/corona_hakab_model/analyzers/example_script.py]: *analysis

File: src/corona_hakab_model/analyzers/population_analysis.py [hunk 4]
```diff
@@ -0,0 +1,85 @@
...
+from typing import Tuple
+import pickle
+import pandas as pd
+from generation.connection_types import ConnectionTypes
+import matplotlib.pyplot as plt
+from analyzers.config import POPULATION_OUTPUT_PATH
+
+
+def load_population_data_to_dfs() -> Tuple[pd.DataFrame, pd.DataFrame, pd.DataFrame]:
+    output_file_path = POPULATION_OUTPUT_PATH
```
Comment (bgmoshe) [inline src/corona_hakab_model/analyzers/population_analysis.py]: Please give the output_file_path
You can give the parameter POPULATION_OUTPUT_PATH as default value

File: src/corona_hakab_model/analyzers/population_analysis.py [hunk 5]
```diff
@@ -0,0 +1,85 @@
...
+    if log_scale:
+        plt.xscale("log")
+
+    ax2 = plt.subplot(212)
+    ax2.plot(relevant_circles.agents_count,
+                relevant_circles.sick_count.divide(relevant_circles.agents_count), "bo")
+    ax2.set_ylim([0, 1.5])
+    ax2.set_title(f"percentage of sick people vs number of agents in {relevant_circles_name}")
+    plt.xlabel(f"number of agents in {relevant_circles_name}")
+    plt.ylabel("number of sick people")
```
Comment (bgmoshe) [inline src/corona_hakab_model/analyzers/population_analysis.py]: You mean percentage and not number, right?

File: src/corona_hakab_model/analyzers/config.py (hunk 1 above)
Comment (mishana) [inline src/corona_hakab_model/analyzers/config.py]: better use:
`from pathlib import Path`
`POPULATION_OUTPUT_PATH =  Path(__file__).parent / "../../../output" `

File: src/corona_hakab_model/analyzers/population_analysis.py [hunk 6]
```diff
@@ -0,0 +1,85 @@
...
+import pickle
+import pandas as pd
+from generation.connection_types import ConnectionTypes
+import matplotlib.pyplot as plt
+from analyzers.config import POPULATION_OUTPUT_PATH
+
+
+def load_population_data_to_dfs() -> Tuple[pd.DataFrame, pd.DataFrame, pd.DataFrame]:
+    output_file_path = POPULATION_OUTPUT_PATH
+    population_data = pickle.load(open(path.join(output_file_path, "population_data.pickle"), "rb"))
```
Comment (mishana) [inline src/corona_hakab_model/analyzers/population_analysis.py]: I think `Path('path_prefix') / 'path_suffix' ` is a better choice throughout the project

**Comment by bgmoshe** [comment]
@NoamShapira Notice that the issue specifies that we want this to run after the simulation ends, so we want this to run as part of main I guess or at least have some sort of CLI interface to it

File: src/corona_hakab_model/analyzers/simulation_analysis.py [hunk 7]
```diff
@@ -18,13 +19,13 @@ def create_comparison_files(files: list = None):
         for each parameter creates a file combining the information from each file given.
     """
     if files is None:
-        files = glob(os.path.join(OUTPUT_FOLDER, "*.csv"))
+        files = glob(os.path.join(TIME_OUTPUT_PATH, "*.csv"))
```
Comment (mishana) [inline src/corona_hakab_model/analyzers/simulation_analysis.py]:
```suggestion
        files = Path(TIME_OUTPUT_PATH).glob("*.csv")
```

File: src/corona_hakab_model/analyzers/simulation_analysis.py [hunk 8]
```diff
@@ -107,10 +108,10 @@ def create_time_step_aggregation(parameters_dir_path, agg_funcs_dict: Dict[str,
     :param agg_funcs_dict: dict of all the aggregation functions that want to be applied across multiple simulation runs
     :return: a dictionary with keys the same as agg_funcs_dict but values the df corresponding to the aggregation
     """
-    files = glob(os.path.join(OUTPUT_FOLDER, parameters_dir_path, "*.csv"))
+    files = glob(os.path.join(TIME_OUTPUT_PATH, parameters_dir_path, "*.csv"))
```
Comment (mishana) [inline src/corona_hakab_model/analyzers/simulation_analysis.py]:
```suggestion
        files = (Path(TIME_OUTPUT_PATH) / parameters_dir_path).glob("*.csv")
```

File: src/corona_hakab_model/analyzers/population_analysis.py (hunk 6 above)
Comment (ghost) [inline src/corona_hakab_model/analyzers/population_analysis.py]: you can use PopulationData's class method import_population_data

File: src/corona_hakab_model/analyzers/population_analysis.py [hunk 9]
```diff
@@ -0,0 +1,85 @@
...
+    ax1.set_title(f"number of sick people vs number of agents in {relevant_circles_name}")
+    plt.xlabel(f"number of agents in {relevant_circles_name}")
+    plt.ylabel("number of sick people")
+    if log_scale:
+        plt.xscale("log")
+
+    ax2 = plt.subplot(212)
+    ax2.plot(relevant_circles.agents_count,
+                relevant_circles.sick_count.divide(relevant_circles.agents_count), "bo")
+    ax2.set_ylim([0, 1.5])
```
Comment (ghost) [inline src/corona_hakab_model/analyzers/population_analysis.py]: y max value isn't 1?

File: src/corona_hakab_model/analyzers/config.py (hunk 1 above)
Comment (NoamShapira) [reply src/corona_hakab_model/analyzers/config.py]: OUTPUT_FOLDER ןis not the correct folder,
we have a open issue to organize tall the outputs from the simulation 2 one place, but right now we do have 2 different output_folders in out repo
so i organized it to one place to allow quick change once the  outputs unified

File: src/corona_hakab_model/analyzers/example_script.py (hunk 2 above)
Comment (NoamShapira) [reply src/corona_hakab_model/analyzers/example_script.py:37]: this is an example for usage
I added, another plot for initial_sick_data

File: src/corona_hakab_model/analyzers/population_analysis.py (hunk 9 above)
Comment (NoamShapira) [reply src/corona_hakab_model/analyzers/population_analysis.py]: yes it just look a bit better with 1.5 as top lim

**Pull request merged**